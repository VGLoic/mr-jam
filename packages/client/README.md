## Available Scripts

In the project directory, you can run:

### `yarn install`

Installs the dependencies for the client application.

### `yarn start:client`

Runs the app in the development mode.<br />
Open [http://localhost:3000](http://localhost:3000) to view it in the browser.

The page will reload if you make edits.<br />
You will also see any lint errors in the console.

You will need the lambda functions running locally in order to use the application. See next section.

### `yarn test`

Launches the test runner for the client application in the interactive watch mode.<br />
See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.

### `yarn test:cover`

Launches the test runner and generates coverage for the client.

### `yarn build`

Builds the client for production to the `packages/client/build` folder.<br />
It correctly bundles React in production mode and optimizes the build for the best performance.

The build is minified and the filenames include the hashes.<br />
Your app is ready to be deployed!

See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.


