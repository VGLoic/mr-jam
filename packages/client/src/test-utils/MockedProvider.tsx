import React from "react";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";
import DateFnsUtils from "@date-io/date-fns";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ethers } from "ethers";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "contexts/theme";
import EthersProvider, { IEthersContext } from "contexts/ethers/EthersProvider";
import TransactionStoreProvider from "contexts/ethers/TransactionStoreProvider";
import CallStoreProvider from "contexts/ethers/CallStoreProvider";

const mockProvider: any = {
  getNetwork: () => 1,
};

const defaultMockEthersContext: IEthersContext = {
  ethereum: {
    isMetaMask: true,
    networkVersion: "1",
    selectedAddress: "0x627306090abaB3A6e1400e9345bC60c78a8BEf57",
    enable: () => Promise.resolve(),
    on: () => {},
    removeListener: () => {},
  },
  isMetaMaskDetected: true,
  isEnabled: true,
  setIsEnabled: () => {},
  isNetworkAllowed: true,
  networkId: "1",
  provider: mockProvider as ethers.providers.Web3Provider,
  selectedAddress: "0x627306090abaB3A6e1400e9345bC60c78a8BEf57",
};

export interface MockedProviderProps {
  children?: React.ReactElement;
  ethersContext?: IEthersContext;
}
const MockedProvider = ({ children, ethersContext }: MockedProviderProps) => {
  const ethersProviderProps: any = {};
  if (ethersContext !== null)
    ethersProviderProps.value = {
      ...defaultMockEthersContext,
      ...ethersContext,
    };

  return (
    <EthersProvider {...ethersProviderProps}>
      <TransactionStoreProvider>
        <CallStoreProvider>
          <Router>
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
              <CssBaseline />
              <ThemeProvider>{children}</ThemeProvider>
            </MuiPickersUtilsProvider>
          </Router>
        </CallStoreProvider>
      </TransactionStoreProvider>
    </EthersProvider>
  );
};

export default MockedProvider;
