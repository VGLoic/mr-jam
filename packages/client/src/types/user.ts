export interface User {
  id: string;
  name: string;
  avatarUrl: string;
}

export interface WrappedUser {
  address: string;
  user: User | null;
}
