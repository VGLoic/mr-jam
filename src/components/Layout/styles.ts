import { makeStyles, Theme } from "@material-ui/core";

const styles = ({ spacing, palette, shadows }: Theme): Styles => ({
  root: {
    display: "flex",
    flexDirection: "column",
  },
  header: {
    height: `56px`,
    background: palette.background.paper,
    paddingLeft: `${spacing(2)}px`,
    paddingRight: `${spacing(2)}px`,
    boxShadow: shadows[3],
    zIndex: 2,
    position: "sticky",
    top: 0,
  },
  content: {
    flex: 1,
    minHeight: `calc(100vh - ${spacing(7)}px)`,
    paddingTop: `${spacing(2)}px`,
    paddingBottom: `${spacing(2)}px`,
    background: palette.background.default,
  },
  itemContent: {
    paddingTop: `${spacing(2)}px`,
    paddingBottom: `${spacing(2)}px`,
  },
});

export interface Styles {
  root: any;
  header: any;
  content: any;
  itemContent: any;
}

export const useStyles = makeStyles(styles);
