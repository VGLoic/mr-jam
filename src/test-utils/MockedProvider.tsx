import React from "react";
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';
import CssBaseline from "@material-ui/core/CssBaseline";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "contexts/theme";


export interface MockedProviderProps {
    children?: React.ReactElement;
}
const MockedProvider = ({ children }: MockedProviderProps) => {
    return (
        <Router>
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
                <CssBaseline />
                <ThemeProvider>
                    {children}
                </ThemeProvider>
            </MuiPickersUtilsProvider>
        </Router>
    )
}

export default MockedProvider;