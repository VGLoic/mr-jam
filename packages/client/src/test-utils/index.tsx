import React from "react";
import { render } from '@testing-library/react'
import MockedProvider from "./MockedProvider";

const customRender = (ui: React.ReactElement, options?: any) => {
  const RenderedMockedProvider = (props: any) => <MockedProvider {...props} ethersContext={options?.ethersContext} />;
  return render(ui, { wrapper: RenderedMockedProvider, ...options });
}

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }