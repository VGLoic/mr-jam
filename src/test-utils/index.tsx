import React from "react";
import { render } from '@testing-library/react'
import MockedProvider from "./MockedProvider";

const customRender = (ui: React.ReactElement, options?: any) =>
  render(ui, { wrapper: MockedProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }