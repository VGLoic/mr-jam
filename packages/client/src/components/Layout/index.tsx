import React from "react";
// UI Components
import { Grid } from "@material-ui/core";
// Component
import Header from "components/Header";
// Styles
import { useStyles } from "./styles";

type LayoutProps = { children: React.ReactNode };
const Layout = ({ children }: LayoutProps) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Header className={classes.header} />
      <main className={classes.content}>
        <Grid container>
          <Grid item md={1} />
          <Grid item md={10} className={classes.itemContent}>
            {children}
          </Grid>
          <Grid item md={1} />
        </Grid>
      </main>
    </div>
  );
};

export default Layout;
