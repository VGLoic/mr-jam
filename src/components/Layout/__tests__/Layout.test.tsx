import React from "react";
import { render, waitFor, screen } from "test-utils";
import Layout from "..";
import { MockedProvider } from '@apollo/client/testing';
import { CurrentUserData } from "components/Header/controllers/currentUser.types";
import { CURRENT_USER } from "components/Header/controllers/currentUser.query";


describe("<Layout />", () => {
    test("it should match snapshot", async () => {
        const currentUserData: CurrentUserData = {
            currentUser: {
                id: "currentUserId",
                name: "Jack Sparrow",
                avatarUrl: "currentUserAvatarUrl"
            }
        };
    
        const currentUserMock = {
            request: {
                query: CURRENT_USER
            },
            result: {
                data: currentUserData
            }
        };

        const { asFragment } = render(
            <MockedProvider mocks={[currentUserMock]}>
                <Layout>
                    <div data-testid="children-test" />
                </Layout>
            </MockedProvider>
        );

        await waitFor(() => screen.getByTestId("user-avatar"))

        expect(asFragment()).toMatchSnapshot("Layout");
    })
})