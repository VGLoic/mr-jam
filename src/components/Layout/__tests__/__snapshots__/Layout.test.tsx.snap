// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> it should match snapshot: Layout 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiGrid-root makeStyles-header-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      data-testid="header-container"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-sm-3"
        data-testid="nyan-cat-animation"
      >
        <img
          alt="nyan-cat"
          class="makeStyles-nyanCat-6"
          src="https://media.giphy.com/media/7lsw8RenVcjCM/giphy.gif"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-9 MuiGrid-grid-sm-6"
      >
        <button
          aria-label="Select a project"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
          data-testid="select-project-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Select a project
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-sm-3"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
        >
          <button
            aria-label="Switch to light mode"
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="toggle-mode-button"
            tabindex="0"
            title="Switch to light mode"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,18V6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,15.31L23.31,12L20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31Z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            aria-label="Open user menu"
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5"
            data-testid="user-avatar"
          >
            <img
              alt="avatar-user"
              class="MuiAvatar-img"
              src="currentUserAvatarUrl"
            />
          </div>
        </div>
      </div>
    </div>
    <main
      class="makeStyles-content-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
        />
        <div
          class="MuiGrid-root makeStyles-itemContent-4 MuiGrid-item MuiGrid-grid-md-10"
        >
          <div
            data-testid="children-test"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
        />
      </div>
    </main>
  </div>
</DocumentFragment>
`;
